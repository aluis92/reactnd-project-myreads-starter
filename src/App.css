.search-books-results {
  padding: 80px 10px 20px;
}

/* book cover */

.book-cover-title {
  padding: 20px 10px 0;
  text-align: center;
  font-size: 0.8em;
}
