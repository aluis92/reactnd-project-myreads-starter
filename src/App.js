import React, { Component } from 'react';
import {
    BrowserRouter,
    Route,
} from 'react-router-dom';
import Home from './pages/Home';
import Search from './pages/Search';
import bookshelfFactory from './utils/bookshelfFactory';

import * as BooksAPI from './services/BooksAPI';

import './App.css';

class App extends Component {
    state = {
        allBooks: [],
        bookList: {
            currentlyReading: [],
            read: [],
            wantToRead: [],
        },
    }

    componentDidMount() {
        this.getAllBooks();
    }

    getAllBooks() {
        BooksAPI.getAll()
            .then(bookshelfFactory)
            .then(bookList => this.setState({ bookList }));
    }

    updateBook = (bookId, shelf) => {
        BooksAPI.update(bookId, shelf)
            .then(() => this.getAllBooks());
    }

    render() {
        const {
            bookList,
        } = this.state;

        return (
            <BrowserRouter>
                <div>
                    <Route
                        exact
                        path="/"
                        render={() => (
                            <Home
                                bookList={bookList}
                                updateBook={this.updateBook}
                            />
                        )}
                    />
                    <Route
                        exact
                        path="/search"
                        component={Search}
                    />
                </div>
            </BrowserRouter>
        );
    }
}

export default App;
