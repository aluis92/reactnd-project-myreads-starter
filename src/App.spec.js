import React from 'react';
import {
    shallow,
    mount,
} from 'enzyme';

import App from './App';

jest.mock('./services/BooksAPI', () => ({
    getAll: () => ({
        then: success => success(
            [
                {
                    id: '1',
                    shelf: 'currentlyReading',
                },
                {
                    id: '2',
                    shelf: 'wantToRead',
                },
                {
                    id: '3',
                    shelf: 'read',
                },
            ],
        ),
    }),
    update: () => ({
        then: (success) => { success(); },
    }),
    search: (query) => {
        let response = null;

        if (query === 'valid query') {
            response = [
                {
                    id: 'searchResp',
                },
            ];
        }

        return ({
            then: (success) => {
                success(response);
            },
        });
    },
}), { virtual: true });

describe('App component', () => {
    const APIResult = [
        {
            id: '1',
            shelf: 'currentlyReading',
        },
        {
            id: '2',
            shelf: 'wantToRead',
        },
        {
            id: '3',
            shelf: 'read',
        },
    ];

    it('should render without crashing', () => {
        expect(shallow(
            <App />,
        )).toMatchSnapshot();
    });


    it('should call getAllBooks when it mounts', () => {
        const spy = jest.spyOn(App.prototype, 'getAllBooks');

        mount(
            <App />,
        );

        expect(spy).toHaveBeenCalled();
    });

    it('should call getAllBooks when it mounts', () => {
        const spy = jest.spyOn(App.prototype, 'getAllBooks');

        mount(
            <App />,
        );

        expect(spy).toHaveBeenCalled();
    });

    it('should have allBooks array state as same as the result of API when it mounts', () => {
        const app = mount(
            <App />,
        );

        expect(app.instance().state.allBooks).toEqual(APIResult);
    });

    it('should have call App´s getAllBooks method when updateBook is triggered', () => {
        const spy = jest.spyOn(App.prototype, 'getAllBooks');

        const app = mount(
            <App />,
        );

        app.instance().updateBook('1', 'read');

        expect(spy).toHaveBeenCalled();
    });

    it('should update App´s searchResults state when searchBooks is called with valid query', () => {
        const app = mount(
            <App />,
        );

        app.instance().searchBooks('valid query');

        expect(app.instance().state.searchResults).toEqual([{ id: 'searchResp' }]);
    });

    it('should set App´s searchResults state with empty array when searchBooks is called with invalid query', () => {
        const app = mount(
            <App />,
        );

        app.instance().searchBooks('invalid query');

        expect(app.instance().state.searchResults).toEqual([]);
    });
});
