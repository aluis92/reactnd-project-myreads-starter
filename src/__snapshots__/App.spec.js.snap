// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without crashing 1`] = `
<BrowserRouter>
  <div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/search"
      render={[Function]}
    />
  </div>
</BrowserRouter>
`;
